[Display_02] Update display comm interface implementation class

jcortes18 opened the pull request:
Closes #163

jcortes18 committed:
Update DisplayCommInterfaceImplementation.java

jcortes18 committed:
Update DisplayCommInterfaceImplementation.java

jcortes18 committed:
Update DisplayCommInterfaceImplementation.java

jcortes18 committed:
Update DisplayCommInterfaceImplementation.java

jcortes18 commented:
I think it's not passing the gradle tests because TrackDataPlotter hasn't been merged into main.

jcortes18 committed:
Update DisplayCommInterfaceImplementation.java

jcortes18 committed:
Update DisplayCommInterfaceImplementation.java

jcortes18 committed:
added toString() methods to LocationDataPoints and TrackData

jcortes18 committed:
Update DisplayCommInterfaceImplementation.java

jcortes18 committed:
Update LocationDataPoint.java

andrewdimmer commented on src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java:
```diff
@@ -14,6 +15,8 @@
 public class DisplayCommInterfaceImplementation implements DisplayCommInterface {
 
   private DisplayComm2WayInterface comm2Way;
+  private TrackData data;
+  private TrackDataPlotter plotter;
```
These don't need to be stored at the class level as they are only used at the method level.
```suggestion
```

andrewdimmer commented on src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java:
```diff
@@ -34,6 +37,17 @@ public DisplayCommInterfaceImplementation(DisplayComm2WayInterface displayComm2W
   */
   public TrackData receiveRfidRequest(int rfidNum) {
-    return comm2Way.passRfidRequest(rfidNum);
+    this.data = comm2Way.passRfidRequest(rfidNum);
```
If we remove the above private class attributes, then this gets the type added to it.
```suggestion
    TrackData data = comm2Way.passRfidRequest(rfidNum);
```

andrewdimmer commented on src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java:
```diff
@@ -34,6 +37,17 @@ public DisplayCommInterfaceImplementation(DisplayComm2WayInterface displayComm2W
+    this.data = comm2Way.passRfidRequest(rfidNum);
+
+    System.out.println("Track Data for RFID tag number " + rfidNum);
+    System.out.println("Location Data Points: " + data.getLocationDataPoints());
+    System.out.println("Course Direction: " + data.getCourseDirection());
+    System.out.println("Speed: " + data.getSpeed());
+    System.out.println("Status: " + data.getStatusMessage());
```
This should be moved to the `toString()` method of TrackData itself. I would also recommend printing the status message first, then the speed and course IF the `TrackData` object is valid, then the points at the end.

andrewdimmer commented on src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java:
```diff
@@ -34,6 +37,17 @@ public DisplayCommInterfaceImplementation(DisplayComm2WayInterface displayComm2W
+    System.out.println("Speed: " + data.getSpeed());
+    System.out.println("Status: " + data.getStatusMessage());
+
+    this.plotter = new TrackDataPlotter(data);
```
If we remove the above private class attributes, then this gets the type added to it.
```suggestion
    TrackDataPlotter plotter = new TrackDataPlotter(data);
```

andrewdimmer commented on src/main/java/edu/oakland/helper/admin/LocationDataPoint.java:
```diff
@@ -69,6 +69,10 @@ public LocalDateTime getTime() {
   }
 
+  public String toString() {
+    return lat + " " + lng + " " + time;
```
I would see if you can make this more descriptive, may something like:
```suggestion
    return "Location Data at " + time.toString() + ":\n\tLatitude: " + lat + "\n\tLongitude: " + lng;
```

andrewdimmer reviewed (changes requested):
Couple of quick changes, then I'd like to take over and add some testing, then you should be good to go!

jcortes18 committed:
Update src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

jcortes18 committed:
Update src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

jcortes18 committed:
Update src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

jcortes18 committed:
Update src/main/java/edu/oakland/helper/admin/LocationDataPoint.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

jcortes18 committed:
update toString methods

jcortes18 committed:
update toString

jcortes18 committed:
Update TrackData.java

andrewdimmer commented on src/main/java/edu/oakland/helper/admin/LocationDataPoint.java:
```diff
@@ -69,6 +69,10 @@ public LocalDateTime getTime() {
     return time;
   }
 
```
I would make sure you add the `@Override notation and a Javadoc
```suggestion
  /**
   * Creates a print-friendly string of the LocationDataPoint data.
   *
   * @return The print friendly string of the LocationDataPoint data.
   */
  @Override
```

andrewdimmer commented on src/main/java/edu/oakland/helper/admin/TrackData.java:
```diff
@@ -123,6 +123,20 @@ public float getSpeed() {
     return points;
   }
 
+  @Override
```
I would add the following Javadoc here:
```suggestion
  /**
   * Creates a print-friendly string of the TrackData data.
   *
   * @return The print friendly string of the TrackData data.
   */
  @Override
```

andrewdimmer commented on src/main/java/edu/oakland/helper/admin/TrackData.java:
```diff
@@ -123,6 +123,20 @@ public float getSpeed() {
+  @Override
+  public String toString() {
+    StringBuilder str = new StringBuilder();
+    if (isValid()) {
+      System.out.println("Status: " + getStatusMessage());
```
It should always print the status, even if it is not valid.
```suggestion
    str.append("Track Data Status: " + getStatusMessage() + "\n");
    if (isValid()) {
```

andrewdimmer commented on src/main/java/edu/oakland/helper/admin/TrackData.java:
```diff
@@ -123,6 +123,20 @@ public float getSpeed() {
+    StringBuilder str = new StringBuilder();
+    if (isValid()) {
+      System.out.println("Status: " + getStatusMessage());
+      System.out.println("Speed: " + getSpeed());
+      System.out.println("Course Direction: " + getCourseDirection());
```
Make sure these get appended to the output!
```suggestion
      str.append("Speed: " + getSpeed() + "\n");
      str.append("Course Direction: " + getCourseDirection() + "\n");
```

andrewdimmer reviewed (changes requested):
Few more minor changes, but once you do this you're all set!

jcortes18 committed:
Update src/main/java/edu/oakland/helper/admin/LocationDataPoint.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

jcortes18 committed:
Update src/main/java/edu/oakland/helper/admin/TrackData.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

jcortes18 committed:
Update src/main/java/edu/oakland/helper/admin/TrackData.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

jcortes18 committed:
Update src/main/java/edu/oakland/helper/admin/TrackData.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

jcortes18 committed:
Update src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

jcortes18 committed:
Update src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

andrewdimmer commented on src/main/java/edu/oakland/helper/admin/LocationDataPoint.java:
```diff
@@ -68,6 +68,15 @@ public float getLng() {
+   *
+   * @return The print friendly string of the LocationDataPoint data.
+   */
+  @Override
```
Oops! My bad... let me fix this one...

jcortes18 committed:
Update src/main/java/edu/oakland/production/display02/DisplayCommInterfaceImplementation.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

andrewdimmer committed:
[Display_02] Add an extra line because I was dumb before

andrewdimmer reviewed (approved):
Looks good to me! I'm going to fix the last of the test issues caused by the Plotter, then this is set to merge!

andrewdimmer committed:
Merge branch 'main' into Update-DisplayCommInterfaceImplementation-Class

jcortes18 committed:
Update DisplayCommInterfaceImplementation.java

jcortes18 committed:
Update DisplayCommInterfaceImplementation.java

andrewdimmer committed:
[Admin] Added LocationDataPoint.toString() Tests

jcortes18 committed:
Merge branch 'Update-DisplayCommInterfaceImplementation-Class' of https://github.com/andrewdimmer/csi-3370-software-project into Update-DisplayCommInterfaceImplementation-Class

jcortes18 committed:
Update DisplayCommInterfaceImplementation.java

andrewdimmer committed:
[Display_02] Expand the TrackDataPlotter to Invalid TrackData

Note that it still does not plot 0 point track data, because that would cause a divide by 0 error.

andrewdimmer committed:
[Display_02] Do NOT Plot 0 Point TrackData in DisplayCommInterface

This prevents a divide by 0 error in the plotter.

andrewdimmer committed:
[Admin] Add TrackData.toString() Tests

Also handle the edge case of no points, and added units to speed and course direction.

andrewdimmer committed:
[Display_02] Fix Checkstyle Test Indentation

andrewdimmer committed:
[Display_02] Fixed Single Point TrackDataPlotter Test

Turns out, it's helpful if you actually use a single point TrackData object to test what happens with a single point TrackData obect...

andrewdimmer committed:
[Display_02] Fixed 2+ Point TrackDataPlotter Test to have 2+ points

andrewdimmer reviewed (approved):
Now, we're all set to merge if everyone else agrees.
